superset/app.py

```diff
@@ -44,5 +52,51 @@ def create_app() -> Flask:
         raise ex
 
 
+def init_config() -> Dict[Any, Any]:
+    config = convert_to_dict(load_default_config())
+    override_conf = convert_to_dict(load_override_config())
+    return merge(config, override_conf)
```

Comment:
I think bringing a new dependency (pydash) just to merge two dicts here is overkill. Why not just use `dict.update`?

I'm also not sure if a recursive merge is the right strategy here. If we have:

```python
# superset/config.py
a = {"b": "c"}
```

And I create a custom config:

```python
# superset_config.py
a = {"d": "e"}
```

This would result in the following configuration:

```python
{'a': {'b': 'c', 'd': 'e'}}
```

Which is probably not what we want.

As a concrete example, we currently have languages configured in `superset/config.py`:

```python
LANGUAGES = {
    "en": {"flag": "us", "name": "English"},
    "es": {"flag": "es", "name": "Spanish"},
    "it": {"flag": "it", "name": "Italian"},
    "fr": {"flag": "fr", "name": "French"},
    "zh": {"flag": "cn", "name": "Chinese"},
    "ja": {"flag": "jp", "name": "Japanese"},
    "de": {"flag": "de", "name": "German"},
    "pt": {"flag": "pt", "name": "Portuguese"},
    "pt_BR": {"flag": "br", "name": "Brazilian Portuguese"},
    "ru": {"flag": "ru", "name": "Russian"},
    "ko": {"flag": "kr", "name": "Korean"},
    "sl": {"flag": "si", "name": "Slovenian"},
}
```

If I want to have just Portuguese in my Superset instance I would add to my config:

```python
LANGUAGES = [
    "pt_BR": {"flag": "br", "name": "Brazilian Portuguese"},
]
```

But the `merge` would bring back all languages.